src/interceptors/ClientRequest/MockHttpSocket.ts

```diff
@@ -272,36 +272,22 @@ export class MockHttpSocket extends MockSocket {
     // if it hasn't been flushed already (e.g. someone started reading request stream).
     this.flushWriteBuffer()
 
-    const httpHeaders: Array<Buffer> = []
-
-    httpHeaders.push(
-      Buffer.from(
-        `HTTP/1.1 ${response.status} ${
-          response.statusText || STATUS_CODES[response.status]
-        }\r\n`
-      )
-    )
+    const w = new Writable({
+      write: (chunk, encoding, callback) => {
+        this.push(chunk, encoding);
+        callback()
+      },
+    })
+    const fakeResponse = new ServerResponse(new IncomingMessage(new net.Socket()))
```

Comment:
Any chance this can also handle compression for us?

Reply:
Hmm.. good point.
I thought we would get this OOTB from Node.js when the response goes back to the client.

Reply:
I hope so! I think we have some tests for compression but I got a new issue about it in MSW recently, which made me think something was still missing. Anyways, we'll see.